app/controllers/users_controller.rb

```diff
@@ -0,0 +1,66 @@
+class UsersController < ApplicationController
+  before_action :authorize_user, only: [:index, :edit, :update, :destroy]
+  before_action :check_current_user,   only: [:show, :edit, :update]
+  before_action :authorize_admin,     only: :destroy
+  before_action :set_user, only: [:edit, :update]
+
+  def show
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      flash[:success] = 'Welcome to the Eureka Caffee!'
+      redirect_to @user
+    else
+      flash.now[:danger] = 'Please verify the information and try again.'
+      render 'new'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
```

Comment:
don't need this, you're setting the user in `set_user`

Reply:
i must've stopped in the middle of making the ```set_user``` and forgot to remove it. removed this time. ✓